### PR TITLE
[sdk] Clean up nullable enable statements

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #3958
Follow-up to #4845

## Changes

* Removed `#nullable enable` from all files in SDK because it is now enabled at the project level.
* Moved `Base2ExponentialBucketHistogram.LowerBoundary.cs` helper out of the SDK into "Shared" folder because it still needs `#nullable enable` for links. Refactored it a bit.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated

### CODE REVIEW

File: test/OpenTelemetry.Tests/MathHelper.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#pragma warning disable SA1119 // Statement should not use unnecessary parenthesis
+
+namespace OpenTelemetry.Tests;
+
+internal static class MathHelper
```
Comment (utpilla) [inline test/OpenTelemetry.Tests/MathHelper.cs:21]: Move this file to the `Shared` folder in `OpenTelemetry.Tests`?

Comment (CodeBlanch) [reply test/OpenTelemetry.Tests/MathHelper.cs]: I doubt this will be used elsewhere but I went ahead and moved it.

Comment (utpilla) [reply test/OpenTelemetry.Tests/MathHelper.cs]: In that case, we could have just kept it in the test file that is using this. Why extract this out into a separate file in the first place?

Comment (CodeBlanch) [reply test/OpenTelemetry.Tests/MathHelper.cs]: Short answer: Clean code!

Long answer: 

> we could have just kept it in the test file that is using this

The thing is, it isn't used by one test file... it is used by two files! If it was just used in the one I would have left it alone for sure. First I made it `internal static` so one of the test classes could call it inside the other. That is a code smell really indicative of essentially a coupling issue due to separation of concerns principle violation. It also has smelly [suppressions for differing style](https://github.com/open-telemetry/opentelemetry-dotnet/blob/b870ed9b0c965ec89bf6b5aedab87ff3cab8ea68/test/OpenTelemetry.Tests/Metrics/Base2ExponentialBucketHistogram.LowerBoundary.Test.cs#L250-L251) from the code copied from runtime. Too many code smells for me I decided to make it clean[er] 😄

**Review by utpilla — APPROVED** [review]

**Pull request merged**